sycl/source/detail/scheduler/graph_builder.cpp

```diff
@@ -69,6 +69,37 @@ static bool isAccessModeAllowed(access::mode Required, access::mode Current) {
   return false;
 }
 
+/// Combines two access modes into a single one that allows both.
+static access::mode combineAccessModes(access::mode A, access::mode B) {
+  if (A == B)
```

Comment:
What about:

{
  if(A == B)
    return A;

  if(A == access::mode::discard_write && (B == access::mode::discard_read_write || B == access::mode::write))
      return B;

  if(B == access::mode::discard_write && (A == access::mode::discard_read_write || A == access::mode::write))
      return A;

  return access::mode::read_write;
}

?

Reply:
That looks much cleaner indeed, applied.